File: Solutions/1108. Defanging an IP Address/1108. Defanging an IP Address.cpp
class Solution {
public:
    string defangIPaddr(string s) {
        string res="";
        for(int i=0;i<s.size();++i){
            if(s[i]=='.')
                res+="[.]";
            else
                res+=s[i];
        }
        return res;
    }
};


class Solution {
public:
    string defangIPaddr(string address) {
        return regex_replace(address, regex("[.]"), "[.]");
    }
};

File: Solutions/1108. Defanging an IP Address/1108. Defanging an IP Address.java
/**
 * 
 */
package github.com.ErnestL1n;

/**
 * @author https://github.com/ErnestL1n
 *
 */
//Leetcode.1108
public class Defanging_an_IP_Address {

	/**
	 * @param args
	 */
	public static String defangIPaddr(String address) {
		StringBuilder str=new StringBuilder();
		for(char c:address.toCharArray()) {
			if(c!='.')str.append(c);
			else str.append("[.]");
		}
		return str.toString();
	}
	public static void main(String[] args) {
		String str=new String("255.100.50.0");
		System.out.println("Input: address = "+str);
		System.out.println("Output: "+defangIPaddr(str));

	}

}
